lua/apisix/balancer.lua

```diff
@@ -151,11 +184,11 @@ local function create_server_picker(upstream, checker)
         end
 
         local picker = resty_chash:new(nodes)
-        local key = upstream.key
         return {
             upstream = upstream,
             get = function (ctx)
-                local id = picker:find(ctx.var[key])
+                local chash_key = create_chash_hash_key(ctx, upstream)
```

Comment:
`fetch_chash_hash_key`

How about this name?

Reply:
ok, is better